src/app/report/edit-display/edit-display.component.ts

```diff
@@ -111,33 +126,59 @@ export class EditDisplayComponent {
   }
 
   closeReport(removeReportFromTree: boolean): void {
+    const node: Report | Checkpoint | undefined = this.selectedNode;
+    if (!ReportUtil.isReport(node)) {
+      this.toastService.showDanger('Could not find report to close');
+      return;
+    }
     this.displayReport = false;
     this.editingRootNode = false;
     this.editingChildNode = false;
     if (removeReportFromTree) {
-      this.closeReportEvent.next(this.report);
+      this.closeReportEvent.next(node);
     }
     this.editor.setNewReport('');
   }
 
   downloadReport(exportBinary: boolean, exportXML: boolean): void {
-    const queryString: string = this.report.xml ? this.report.storageId.toString() : this.report.uid.split('#')[0];
+    const node: Report | Checkpoint | undefined = this.selectedNode;
+    if (!node) {
+      this.toastService.showDanger('Could not find report to download');
+      return;
+    }
+    let queryString: string;
+    if (ReportUtil.isReport(node)) {
+      queryString = String(node.storageId);
+    } else if (ReportUtil.isCheckPoint(node)) {
+      queryString = node.uid.split('#')[0];
+    } else {
+      queryString = '';
+    }
+    if (!queryString) {
+      this.toastService.showDanger('Could not find report to download');
+      return;
+    }
     this.helperService.download(`${queryString}&`, this.currentView.storageName, exportBinary, exportXML);
     this.httpService.handleSuccess('Report Downloaded!');
   }
 
   selectStubStrategy(event: Event): void {
-    this.saveChanges((event.target as HTMLOptionElement).value);
+    this.saveChanges(Number.parseInt((event.target as HTMLOptionElement).value));
   }
 
   openDifferenceModal(type: ChangesAction): void {
+    const node: Report | Checkpoint | undefined = this.selectedNode;
+    if (!node) {
+      this.toastService.showDanger('Could not find report to open difference modal');
+      return;
+    }
```

Comment:
I see these two lines used in multiple methods, maybe it's better to move this to a method that just returns a checkpoint or a report and shows a toast if it's null or undefined?

Reply:
That's a good idea. I'll add the change.

Reply:
Added the change